src/VsExtension/NuGetPackage.cs

```diff
@@ -372,6 +368,52 @@ private void SetDefaultCredentialProvider()
                 // add the proxy to v2 proxy cache.
                 v2ProxyCache?.Add(proxy);
             };
+
+            NuGet.Protocol.Core.v3.HttpHandlerResourceV3.ProxyPassed = proxy =>
+            {
+                // add the proxy to v2 proxy cache.
+                v2ProxyCache?.Add(proxy);
+            };
+
+            var v2CredentialStoreType = typeof(NuGet.ICredentialCache).Assembly.GetType("NuGet.CredentialStore");
```

Comment:
The type is `Internal` - `ProxyCache` used to be internal but was changed in 2.8 (hence the code change above). This one continues to be internal.